pyomo/contrib/gdpopt/GDPopt.py

```diff
@@ -287,7 +287,7 @@ def solve(self, model, **kwds):
         solve_data.results.problem.lower_bound = solve_data.LB
         solve_data.results.problem.upper_bound = solve_data.UB
 
-        if config.tee:
+        if config.tee and old_logger_level > logging.INFO:
             config.logger.setLevel(old_logger_level)
```

Comment:
There is a subtle issue here where an exception raised by GDPopt will cause the logger level to not be restored.  A slightly more robust approach would be:
```
old_logger_level = config.logger.getEffectiveLevel()
try:
    if config.tee and old_logger_level > logging.INFO:
        # If the logger does not already include INFO, include it.
        config.logger.setLevel(logging.INFO)
    # ........
finally:
    # No need to test tee ... just unconditionally restore the log level.
    config.logger.setLevel(old_logger_level)
```